packages/components/src/sites-table/use-sites-table-sorting.tsx

```diff
@@ -0,0 +1,35 @@
+// eslint-disable-next-line no-restricted-imports
+import type { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+
+interface SitesTableSortOptions {
+	sort?: string;
+	order?: string;
+}
+
+interface UseSitesTableSortingResult {
+	sortedSites: SiteExcerptData[];
+}
+
+export function useSitesTableSorting(
+	allSites: SiteExcerptData[],
+	{ sort, order = 'asc' }: SitesTableSortOptions
+): UseSitesTableSortingResult {
+	switch ( sort ) {
+		case 'updated-at':
+			return { sortedSites: sortSitesByLastPublish( allSites, order ) };
+		default:
+			return { sortedSites: allSites };
+	}
+}
+
+function sortSitesByLastPublish( sites: SiteExcerptData[], order: string ): SiteExcerptData[] {
+	return sites.sort( ( a, b ) => {
+		if ( a.options?.updated_at && b.options?.updated_at ) {
+			return (
+				( Date.parse( a.options?.updated_at ) - Date.parse( b.options?.updated_at ) ) *
```

Comment:
The `updated_at` field uses the `YYYY-MM-DDTHH:MM:SS` format, which means it's sortable using regular `<` or `>` comparisons. I like the trick of multiplying by `1` or `-1`, but I suspect that calling the `Date.parse()` function is much more expensive than doing string comparisons.

Also, not a biggy, but the `?` aren't necessary here because we've already checked whether the property exists in the `if` statement above, and TypeScript is clever enough to figure that out.

```suggestion
				( Date.parse( a.options.updated_at ) - Date.parse( b.options.updated_at ) ) *
```

Reply:
@p-jackson, thanks for the suggestions. I've updated the sorting implementation to use string comparison.

Reply:
Probably only thought of it because of today's discussion on dates 😸  p1658927993059409-slack-C025Q5RK2
I'm not sure we have to account for leap-second smearing though 🤔